gst/nnstreamer/include/tensor_typedef.h

```diff
@@ -43,6 +43,12 @@
  */
 #define NNS_TENSOR_META_RANK_LIMIT	(16)
 
+/**
+ * @brief The transpose rank is fixed to 3.
+ * This RANK does not affect other/tensors(s)'s NNS_TENSOR_RANK_LIMIT.
+ */
+#define NNS_TENSOR_TRANSPOSE_RANK_LIMIT (3)
```

Comment:
1. 3.. Not 4?
2. Having information depending on the implementation of "transpose" at a common module is not a good design (circular dependency). This macro should be defined in transform context only.

Reply:
You'll need an exception and error handling for tensor-transform/transpose when users have requested some op w/ RANK_LIMIT > rank > TRANSPOSE_RANK_LIMIT